best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,285 @@
+
+- a Node using the IS-04 Registration API
+- a monitoring application using the IS-04 Query API
+- a connection control application using the IS-05 Connection API
+
+## Introduction (informative)
+
+This document covers the automated provisioning of TLS Certificates to NMOS Servers, which are then used to secure communications between NMOS Servers and NMOS Clients.
+
+This document is not concerned with the security of the connection used to carry out provisioning of the TLS Certificate, but for the mechanism described in this document to be effective the connection must be secured, ideally using the recommendation covered in [BCP-003-01](best-practice-secure-comms.md).
+
+## Automated Certificate Provisioning Flow (informative)
+
+To enable zero-configuration TLS Certificate provisioning, manufacturers must include a unique TLS Client Certificate on all their devices and the provide the corresponding Root Certificate Authority to customers. If the TLS Client Certificate or the chain of trust is compromised the manufacturer must revoke the comprised certificates.
+
+1. Before the NMOS Node(EST Client) is shipped from the factory it must be provisioned with a unique TLS Client Certificate, signed by the Manufacturers Certificate Authority
+2. When the EST Client is connected to the target environments network, it will first discover the location of the EST Server using Unicast DNS-SD.
+    * The EST Client should assume the EST server found using DNS-SD is trusted
+3. The EST Client should then request the Root CA for the target network, from the EST Server
+    * Using the Root CA returned to secure further communications with the EST Server and NMOS Servers
+4. The EST Client must generate a Certificate Signing Request(CSR) for all supported TLS Cipher Suites with appropriate key lengths
+    * Included in the CSR must be the DNS resolvable name for the current domain of the EST Client
+    * Appropriate values for the remaining CSR fields should be used
+5. The EST Client should send each CSR to the EST Server
+    * The EST Client must provide the Manufacturer issued certificate during the TLS handshake, if valid
+    * The EST Server will use the TLS Client Certificate to validate whether the EST Client is authorized to be issued with a TLS Server Certificate. If the EST CLient is validated the EST Server will return a signed TLS Certificate
+6. The EST Client should provide the returned TLS Certificate and the corresponding chain of trust for all future request to its NMOS APIs
+
+The workflow if no valid Manufacturer TLS Client Certificate is present or if EST Client is moved to a new network are covered in detail in the following [sections](#est-client).
+
+## DNS-SD Advertisement
+
+The EST Server MUST be advertised using unicast DNS-SD as per [RFC 6763][RFC-6763].
+The EST Server SHOULD NOT be advertised via mDNS-based DNS-SD.
+EST Clients MUST not trust a mDNS-based DNS-SD advertisement for the EST server.
```

Comment:
Is there a specific reason for not trusting the mDNS one, other than that it's easier to spoof? Perhaps it should not 'automatically trust' these, or like other specs if a unicast record is available mDNS ones should be ignored?

Reply:
Yes I thought it best not to use mDNS announcements as they can be easily spoofed and there is to be no TLS Authentication to be performed  when the EST Client first communicates with the EST server. 
You wouldn't for example want a device to be successfully enrolled in a network, discover a higher priority EST Server when renewing the certificate, and get the Root CA and TLS Certificate of a rogue CA and stop working. The query/Registration APIs being the biggest risk here, as they would no longer be trusted by Nodes.
This could be a configurable option on the EST Client?